AlphaWallet/alpha-wallet-ios
# #4507 add select network screen after you create import a new wallet

Closes #4507.

I will squish and merge once approved.

##### Create Wallet Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect InitialNetworkSelectionViewController (screen 1) to show up.
5. Select one or more networks in main net.
6. Press continue button.
7. Expect TokensViewController (screen 6) to show up.

##### Import Wallet Test:
1. Delete the app from your device.
2. Run.
3. Select "I already have a wallet" (screen 7).
4. Select "Import existing wallet" (screen 8).
5. Enter wallet data.
6. Expect InitialNetworkSelectionViewController (screen 1) to show up.
7. Select one or more networks in mainnets.
8. Press continue button.
9. Expect TokensViewController (screen 6) to show up. 

#### Watch Wallet Test:
1. Delete the app from your device.
2. Run.
3. Select "I already have a wallet" (screen 7).
4. Select "Track wallet address" (screen 8).
5. Enter wallet address.
6. Expect TokensViewController (screen 6) to show up. 

#### Testnet Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect to see InitialNetworkSelectionViewController (screen 1).
5. Click on the Mainnet toggle switch to turn off mainnets.
6. Expect Overlay to appear (screen 3).
7. Press "Got it, enable testnets" button.
8. Expect Testnets screen (screen 4).
9. Click on the Testnet toggle switch to turn off testnets.
10. Expect to see InitialNetworkSelectionViewController (screen 1).

#### Filter Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect to see InitialNetworkSelectionViewController (screen 1).
5. Select the searchBar and type some letters.
6. Expect to see searchBar filtering (screen 2) or empty search results (screen 9).

#### Continue button Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect to see InitialNetworkSelectionViewController (screen 1).
5. Deselect all networks.
6. Expect to see Continue button disabled (screen 5)

| Screen # | 
| ----- |
| ![Screen 1](https://user-images.githubusercontent.com/1050309/168063954-5a1e27f4-d692-4558-be46-6f74023ec2ed.png)<BR>Screen 1<BR>InitialNetworkSelectionViewController |
| ![2](https://user-images.githubusercontent.com/1050309/168064528-874d483a-0ac4-484c-a8f6-fb143d171d29.png)<BR>Screen 2<BR>SearchBar filtering |
| ![3](https://user-images.githubusercontent.com/1050309/168064726-1033fb71-7e2d-4cdb-bade-98aa29a94c46.png)<BR>Screen 3<BR>Testnets Overlay |
| ![4](https://user-images.githubusercontent.com/1050309/168065393-199db8e3-0a73-4d0d-82cb-2e803acb9a88.png)<BR>Screen 4<BR>Testnets |
| ![5](https://user-images.githubusercontent.com/1050309/168065699-b71e2444-eb71-4330-a445-695cefed0222.png)<BR>Screen 5<BR>Continue button disabled when no networks are selected |
| ![6](https://user-images.githubusercontent.com/1050309/168066150-61c14f19-6876-4a5b-b003-ff4cbb6616a9.png)<BR>Screen 6<BR>TokensViewController |
| ![7](https://user-images.githubusercontent.com/1050309/168066696-99b82bd3-55a0-48d2-9bcd-cd7e9a4b28f7.png)<BR>Screen 7<BR>Initial Wallet Creation |
| ![8](https://user-images.githubusercontent.com/1050309/168067165-3d8feed1-629f-487e-8fe4-2b3d16b097e0.png)<BR>Screen 8<BR>Import/Track Wallet Options |
| ![9](https://user-images.githubusercontent.com/1050309/168281987-33d9ba46-4204-48d7-8273-e95ef3eef1cc.png)<BR>Screen 9<BR>Empty search results |
